README.md
# surfinmd
Binding **Surf**ace **in** **M**olecular **D**ynamics

SurfinMD computes the binding surface area between a pair of chains as indicated in a reference structures.
In addition it decomposes the binding area by residue, allowing the user to dig deep into the interactions.

There is an extensive documentation produced by Doxygen. Download and have fun browsing it :) 

```bash
###################################################
 # Program:  SurfinMD - version 1.06p4 - 14Oct2016 #
 # Diego E.B. Gomes(1), Gabriel Limaverde SCS(2)   #
 # Alan Wilter SS (3), Pedro G. Pascutti(4)        #
 # 1) INMETRO - Brasil                             #
 # 2) Fundacao Oswaldo Cruz - Brasil               #
 # 3) EMBL - EBI, Wellcome Trust Genome Campus     #
 # 4) Universidade Federal do Rio de Janeiro       #
 # mailto: dgomes@pq.cnpq.br                       #
 ###################################################

 Usage: 

 surfinmd -r ref.pdb -i traj.pdb -dcd traj.dcd -o surf.dat -or rsurf.dat -csv rsurf.csv -p 1.4 -d 1.0 -a 6

 Please review all usage options [defaul]

  -h    = Display this help 
  -r    = Reference Structure [ref.pdb]
  -i    = Input file          [traj.pdb]
  -dcd  = Input file          [traj.dcd]
  -o    = Output surface      [surf.dat]
  -or   = Output surface/residue [rsurf.dat]
  -csv  = Output ALL surf/res [rsurf.csv]
  -p    = probe radius        [1.4]
  -d    = probe density       [1.0 A^2]
  -a    = attention number    [6]
  -ext  = (Rvdw + ext )       [n]
  -scal = (Rvdw * scal)       [n]


LMDM - SURFINMD - version 1.06 - Fri Nov 20 11:38:58 BRST 2015

Developed by:
!debg           ; Diego Enry B. Gomes,                  ; dgomes@pq.cnpq.br
!awss           ; Alan Wilter                           ; 
!gabriel        ; Gabriel Limaverde S. C. Sousa         ; 
!melomcr        ; Marcelo C. R. Melo                    ; 
!pegepa         ; Pedro G. Pascutti                     ; pascutti@biof.ufrj.br

Laboratorio de Modelagem e Dinamica Molecular
Instituto de Biofísica Carlos Chagas Filho - UFRJ


# Install
1. Edit "complile.sh" to select your favorite, and available compiler (intel is the fastest):
"g77,gfortran,intel32,intel64,sun"
compiler=intel64

2. If necessary, change the compiler path and flags

3. Run the compilation script " ./compile".

II) Usage
1. You need a PDB file with TWO diffent chain identifiers: A for the receptor and B for the ligand.
2. Parameters are passed through the command line.
3. Type "surfinmd -h" for more information.

WARNING: You MUST have only two chains: A and B !!!

III) Test

cd test/
./please.test


----//-----
Compared to PISA on 1BVE.PDB  (I think it gives you the average of all frames)
chain1    chain2    area/a^2
   A          B       1583.8
   B         DMP       354.7
   A         DMP       348.2
Sum of chain A and B chains with DMP = 702.9


SurfinMD results
gfortran |  Runtime =   8.588537     seconds
 model           1  total area   654.5414
 model           2  total area   620.0327
 model           3  total area   694.4353
 model           4  total area   673.0942
 model           5  total area   697.4927
 model           6  total area   698.9967
 model           7  total area   731.1608
 model           8  total area   682.5359
 model           9  total area   727.3707
 model          10  total area   648.9081
 model          11  total area   726.0377
 model          12  total area   692.0990
 model          13  total area   675.2982
 model          14  total area   685.4004
 model          15  total area   658.2787
```


CHANGELOG.txt
!pegepa         ; Pedro G. Pascutti
!awss           ; Alan Wilter
!debg / dgomes  ; Diego Enry B. Gomes, 
!gabriel        ; Gabriel Limaverde S. C. Sousa
!melomcr        ; Marcelo C. R. Melo


***********************************************************************
Todo list
***********************************************************************
1) Ainda não faz superfície molecular, só faz a intermolecular.
2) A área por átomo passa por vetor APENAS para "atten=6" as demais passam via dots e NÃO são suportadas por esta versão.

Aqui está a modificação que permite usar um vetor para guardar os dados que iriam para o "dots".
É necessário modificar o "write(7,550)..." para todas as "atten", para "atmarea(i2)=atmarea(i2)+area".


**********************************************************************
versao 1.06p1 - Mon Nov 23 21:19:54 BRST 2015
**********************************************************************
1) Atualizei o jeito como o SURF imprime os resultados, agora a area do receptor e ligante sao escritas separadamente.
  Por que o Ligante (nao proteico) nao tem definicao do que eh Hidrofobico e Hidrofilica.
2) Inclui novos residuos na tabela de Hidrofobicos e Hidrofilicos... ainda precisamos completar com os nomes de todos os FF.
3) Criei uma pasta "tools" para deixar scripts para analises multiplas. (cria painel xmgrace)
4) Pequenos ajustes de identacao no codigo do lmdm.mds.f


**********************************************************************
versao 1.06 - Fri Nov 20 12:00:08 BRST 2015
**********************************************************************

1) Incluido o suporte a arquivos DCD.
2) Com o DCD é possivel selecionar trechos da trajetoria com as flags "-b” e “-e”
3) Algumas limpezas no codigo
4) Documentacao com o DOXYGEN (vejam o HTML na pasta doc ) 

OBS o suporte a NetCDF e XTC precisa compilar bibliotecas externas. (Já consegui um SURF-beta com netcdf, só não tenho um jeito simples de instalar)


**********************************************************************
versao 1.05.r6 
O samuel me enviou uns arquivos em que o surf nao produz area para o ligante
estou trabalhando neles para resolver.
O que acontece eh que se o numero dos residuos de aminoacidos nao eh continuo o amnmed(i), amnarea(i) 
pega o numero que estiver la, e nao o numero "serial". Entao mais para a frente o programa se nao escreve todas as areas:

ex:
tem 5 residuos, mas a numeracao esta assim:
1 2 3 4 9
o prog plota as areas de 1, 2, 3, 4, 5  (cade o 9 ?). entendeu ?

modificacao no surfinmd.f e getarea.f !bugfix v1.05r6

**********************************************************************
versao 1.05.r5 eu nao lembro qq mudei


**********************************************************************
versao 1.05.r4 | dgomes, Tue Feb  8 00:08:03 CET 2011
Feature request. O Pedro Valiente pediu uma alteracao para poder rodar mais de 100mil frames
Agora a versao "BIG" nao calcula o desvio padrao, no entanto consegue calcular quantos frames voce quiser.
Ainda tenho minhas duvidas quanto ao limite disso por que eh preciso armazenar a soma de todas as areas de contato
nao sei se o computador vai aguentar um numero tao grande.

***********************************************************************
versao 1.05.r3 |
Bugfix na versao do Valiente. Agora suporta 50mil atomos.

***********************************************************************
versao 1.05.r2 |
Adicionado a compilacao especial para a Priscila S.F. Celestino e Pedro Valiente F.

***********************************************************************
versao 1.05r1 | 
***********************************************************************
Corrigido. O parametro MAXNBR nao havia sido mudado em todos os lugares.
O resultado agora bate com a versao v1.04 e eh parecido com o PISA para 1BVE.PDB (pasta ./test)

#readref.f  Modificado para lidar com arquivos de referencia com estes finais:
          if((line(1:6).eq."TER   ").or.(line(1:6).eq."ENDMDL").
     & or.(line(1:6).eq."      ").or.(line(1:6).eq."END   ")) then


***********************************************************************
versao 1.05 | Wed May 27 09:41:32 PDT 2009 !by debg
***********************************************************************
Adicionado o valor para o atomo P3 do martini forcefield. 
Existe alguma coisa errada com esta versao pois o resultado esta dando 
a metade do valor de referencia produzido pelo PISA. Mon Feb 22 18:42:48 BRT 2010
Adicionada a rotina para "getatype.f" para separar a area por tipo de residuo. (total, hidrofobico, hidrofilico)

***********************************************************************
versao 1.04 | Tue Jul 29 22:04:21 BRT 2008 !by debg
***********************************************************************
1) Organizei o código como o Alan pediu.

2) Ainda NÃO corrigi a limitação do "atten=6"

3) Comecei a adicionar códigos de erro nos arquivos fonte, assim fica mais 
fácil para identificar problemas na execução do programa. Procure por
!error codes em todos os arquivos fonte.

A dica para criar esses códigos de erro é terminar o programa ou as subrotinas
com "return" e "end" 

ex.

return
!error codes
999     write(*,*) "ERRO: Arquivo não encontrado."
end

4) O código com o maior número de modificações foi o readparam.f
Agora não é mais necessário ter o arquivo "surfinmd.par" formatado certinho
de um jeito que o "readparam.f" antigo conseguisse ler. 
Os parâmetros do SURFINMD agora são passados via linha de comando (igual
ao GROMACS :D ).
Nem parece mas deu um baita trabalho para fazer o programa entender direito
esses argumentos, independente da ordem e sem dar erro e fazer CHARACTER
virar INTEGER ou REAL... enfim esse trabalho todo para fazer algo como o
exemplo abaixo.

          if(argv(1:4).eq."-ext") then ; call getarg(i+1,argv)
            if(argv.eq." ".or.argv(1:1).eq."-") argv=default(7)
            read(argv,*,err=950) exclude !string to INTEGER
          endif


src/help.f
************************************************************************
        subroutine help()
************************************************************************
        character red*5,normal*5,underline*4

        red=char(27)//"[31m"
        normal=char(27)//"[0m"
        underline=char(27)//"[4m"
       print*
      write(*,*) underline//"Usage:"//normal
       print*

       write(*,*) "surfinmd"
     &  //red//" -r"//normal//"ref.pdb"
     &  //red//" -i"//normal//"traj.pdb"
     &  //red//" -dcd"//normal//"traj.dcd"
     &  //red//" -o"//normal//"surf.dat"
     &  //red//" -or"//normal//"rsurf.dat"
     &  //red//" -csv"//normal//"rsurf.csv"
     &  //red//" -p"//normal//"1.4"
     &  //red//" -d"//normal//"1.0"
     &  //red//" -a"//normal//"6"

      print*
      write(*,*) 
     & underline//"Please review all usage options"//normal//"[defaul]"
      print*


        write(*,*)
     & red//" -h   "//normal//"= Display this help "
        write(*,*)
     & red//" -r   "//normal//"= Reference Structure [ref.pdb]"
        write(*,*)
     & red//" -i   "//normal//"= Input file          [traj.pdb]"
        write(*,*) 
     & red//" -dcd "//normal//"= Input file          [traj.dcd]"
        write(*,*)
     & red//" -o   "//normal//"= Output surface      [surf.dat]"
        write(*,*)
     & red//" -or  "//normal//"= Output surface/residue [rsurf.dat]"
        write(*,*)
     & red//" -csv "//normal//"= Output ALL surf/res [rsurf.csv]"
        write(*,*) 
     & red//" -p   "//normal//"= probe radius        [1.4]"
        write(*,*) 
     & red//" -d   "//normal//"= probe density       [1.0 A^2]"
        write(*,*) 
     & red//" -a   "//normal//"= attention number    [6]"
        write(*,*) 
     & red//" -ext "//normal//"= (Rvdw + ext )       [n]"
        write(*,*) 
     & red//" -scal"//normal//"= (Rvdw * scal)       [n]"


        print*
        STOP
        return
        end

